### PR TITLE
Update to glog v1.2.2

### PR DESCRIPTION
**Pull request by liujed**
Updated to glog v1.2.2 and reimplemented #1. Upstream has changed considerably since v1.0.0.

### CODE REVIEW

**Comment by liujed** [comment]
My changes here are in 77128ec, which reverts #1; and in a2fa19f, which re-implements #1 on top of glog v1.2.2.

File: internal/logsink/logsink.go [hunk 1]
```diff
@@ -244,6 +248,13 @@ func textPrintf(m *Meta, textSinks []Text, format string, args ...any) (n int, e
 	}
 	buf.WriteString("] ")
 
+	if m.Prefix != nil {
+		buf.WriteString(*m.Prefix)
+		if !strings.HasSuffix(*m.Prefix, " ") {
+			buf.WriteString(" ")
+		}
```
Comment (mgritter) [inline internal/logsink/logsink.go:255]: I think it makes more sense to do this once in SetLogPrefix, as the original did.

**Review by mgritter — APPROVED** [review]
My brief look at the new structure suggests that textPrintf is an appropriate place.

I wonder whether this requirement could now be met by configuring an appropriate sink, like the example provided for StructuredTextWrapper: https://github.com/golang/glog/blob/97303146a4ffecf364d4300e07fca855d0062c43/internal/logsink/logsink.go#L377

which would prevent us from having a custom glog at all.  I didn't think there was a layer of indirection available earlier.  But, this example seems to involve a lot of mucking abaout in "internal" so I don't see how it would work in practice.

File: internal/logsink/logsink.go (hunk 1 above)
Comment (liujed) [reply internal/logsink/logsink.go:255]: The space separator is not necessarily desired for structured sinks, which may choose to put the prefix in its own field.

**Pull request merged**